File: Jaydeep/DNN Training/bhattji Modules/bhattji_XYZ70s/Settings.ascx.cs
// Copyright (c) 2002-2005
// by Jaydeep Bhatt, Vision Consultants. ( http://www.bhattji.com )
//
// Permission is hereby granted, to the person obtaining a copy of this software legaly and associated 
// documentation files (the "Software"), to deal in the Software with restriction, including with limitation 
// the rights to use, copy, modify, merge, PublishDate, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//
using System;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using DotNetNuke;
using DotNetNuke.Common;
using DotNetNuke.Services.Localization;
using DotNetNuke.Services.Exceptions;
//using bhattji.Modules.XYZ70s.Business;
using DotNetNuke.Entities.Modules;
using DotNetNuke.Entities.Users;

namespace bhattji.Modules.XYZ70s
{
    public abstract partial class Settings : XYZ70SettingsBase
	{

		#region " Methods "

		private void SetIconBar()
		{
			//Give the ImageUrl
            //imbUpdate.ImageUrl = ControlPath + "img/bhattji_Update.jpg";
            //imbCancel.ImageUrl = ControlPath + "img/bhattji_Cancel.jpg";

			//Give the Tooltip
			cmdUpdate.ToolTip = Localization.GetString("cmdUpdate");
			//cmdUpdate.Text
			cmdCancel.ToolTip = Localization.GetString("cmdCancel");
			//cmdCancel.Text
		}
		//SetIconBar()

		private void OptionsToPage(int ModId)
		{
			if (ModId > 0) OptionsToPage(new OptionsInfo(ModId)); 
		}

		private void OptionsToPage(OptionsInfo objOI)
		{
			{
				//Main Options
				try {
					rblItemsScope.SelectedIndex = objOI.ItemsScope;
				}
				catch {
					rblItemsScope.SelectedIndex = 0;
				}
				try {
					rblCategoriesScope.SelectedIndex = objOI.CategoriesScope;
				}
				catch {
					rblCategoriesScope.SelectedIndex = 2;
				}
				chkOnlyHostCanEdit.Checked = objOI.OnlyHostCanEdit;
				try {
					rblViewControl.Items.FindByValue(objOI.ViewControl).Selected = true;
				}
				catch {
					rblViewControl.SelectedIndex = 0;
				}
				chkHideSearch.Checked = objOI.HideSearch;
				txtPagerSize.Text = objOI.PagerSize.ToString();
				txtNoOfPages.Text = objOI.NoOfPages.ToString();
				chkDeleteFromGrid.Checked = objOI.DeleteFromGrid;

				try {
					ddlUpdateRedirection.Items.FindByValue(objOI.UpdateRedirection).Selected = true;
				}
				catch {
				}

				//Option1 Options
				chkAddDescription.Checked = objOI.AddDescription;
				try {
					rblTabCss.Items.FindByValue(objOI.TabCss).Selected = true;
				}
				catch {
					rblTabCss.SelectedIndex = 0;
				}
				try {
					ddlTransition.Items.FindByValue(objOI.Transition).Selected = true;
				}
				catch {
					ddlTransition.Items.FindByValue("Pixelate").Selected = true;
				}
				chkRattleImage.Checked = objOI.RattleImage;
				try {
					ddlAddDate.Items.FindByValue(objOI.AddDate).Selected = true;
				}
				catch {
				}
				try {
					ddlImagePosition.Items.FindByValue(objOI.ImagePosition).Selected = true;
				}
				catch {
					ddlImagePosition.SelectedIndex = 1;
				}
				chkDynamicThumb.Checked = objOI.DynamicThumb;
				chkShowRatings.Checked = objOI.ShowRatings;
				txtThumbWidth.Text = objOI.ThumbWidth;
				txtThumbHeight.Text = objOI.ThumbHeight;
                txtHighLightColor.Text = objOI.HighLightColor;
                txtGridCss.Text=objOI.GridCss;
                txtHeaderCss.Text =objOI.HeaderCss;
                txtRowCss.Text=objOI.RowCss;
                txtAltRowCss.Text=objOI.AltRowCss;
                txtSelRowCss.Text=objOI.SelRowCss;
                txtEditRowCss.Text=objOI.EditRowCss;
                txtPagerCss.Text=objOI.PagerCss;
                txtFooterCss.Text=objOI.FooterCss;
                //txtBackColor.Text = objOI.BackColor;
                //txtAltColor.Text = objOI.AltColor;
                //txtSelectedColor.Text = objOI.SelectedColor;
                //txtHeaderBackColor.Text = objOI.HeaderBackColor;
                //txtPagerBackColor.Text = objOI.PagerBackColor;
				txtThumbColumns.Text = objOI.ThumbColumns.ToString();
                txtThumbRows.Text = objOI.ThumbRows.ToString();
				chkHideTextLink.Checked = objOI.HideTextLink;

				//Scroller Options
				try {
					rblScrollBehavior.Items.FindByValue(objOI.ScrollBehavior).Selected = true;
				}
				catch {
					rblScrollBehavior.SelectedIndex = 0;
				}
				try {
					rblScrollDirection.Items.FindByValue(objOI.ScrollDirection).Selected = true;
				}
				catch {
					rblScrollDirection.SelectedIndex = 0;
				}
				txtScrollAmount.Text = objOI.ScrollAmount;
				txtScrollDelay.Text = objOI.ScrollDelay;
				txtScrollWidth.Text = objOI.ScrollWidth;
				txtScrollHeight.Text = objOI.ScrollHeight;

				//Audit Control
				try {
					ctlAudit.CreatedByUser = objOI.UpdatedByUser;
                    ctlAudit.CreatedDate = objOI.UpdatedDate.ToString();
                    //UserInfo objHostUser = UserController.GetUser(PortalId, UserId, true);
                    UserInfo objHostUser = UserController.GetUserById(PortalId, UserId);
					rblItemsScope.Enabled = objHostUser.IsSuperUser;
					rblCategoriesScope.Enabled = objHostUser.IsSuperUser;
					chkOnlyHostCanEdit.Enabled = objHostUser.IsSuperUser;
					if (objOI.OnlyHostCanEdit)
					{
						//txtPayPalBusinessID.Enabled = objHostUser.IsSuperUser
					}
				}
				catch {
					ctlAudit.Visible = false;
				}
			}

		}

		private void PageToOptions(int ModId)
		{
			OptionsInfo objOI = new OptionsInfo();
			{
				//Main Options
				objOI.ItemsScope = rblItemsScope.SelectedIndex;
				objOI.CategoriesScope = rblCategoriesScope.SelectedIndex;
				objOI.OnlyHostCanEdit = chkOnlyHostCanEdit.Checked;
				objOI.ViewControl = rblViewControl.SelectedValue;
				objOI.HideSearch = chkHideSearch.Checked;
				try {
					objOI.PagerSize = int.Parse(txtPagerSize.Text);
				}
				catch {
				}
				try {
					objOI.NoOfPages = int.Parse(txtNoOfPages.Text);
				}
				catch {
				}
				objOI.DeleteFromGrid = chkDeleteFromGrid.Checked;
				objOI.UpdateRedirection = ddlUpdateRedirection.SelectedValue;

				//Option1 Options
				objOI.AddDescription = chkAddDescription.Checked;
				objOI.TabCss = rblTabCss.SelectedValue;
				objOI.Transition = ddlTransition.SelectedValue;
				objOI.RattleImage = chkRattleImage.Checked;
				objOI.AddDate = ddlAddDate.SelectedValue;
				objOI.ImagePosition = ddlImagePosition.SelectedValue;
				objOI.DynamicThumb = chkDynamicThumb.Checked;
				objOI.ShowRatings = chkShowRatings.Checked;
				objOI.ThumbWidth = txtThumbWidth.Text;
				objOI.ThumbHeight = txtThumbHeight.Text;
                objOI.HighLightColor = txtHighLightColor.Text;
                objOI.GridCss = txtGridCss.Text;
                objOI.HeaderCss = txtHeaderCss.Text;
                objOI.RowCss = txtRowCss.Text;
                objOI.AltRowCss = txtAltRowCss.Text;
                objOI.SelRowCss = txtSelRowCss.Text;
                objOI.EditRowCss = txtEditRowCss.Text;
                objOI.PagerCss = txtPagerCss.Text;
                objOI.FooterCss = txtFooterCss.Text;
                //objOI.BackColor = txtBackColor.Text;
                //objOI.AltColor = txtAltColor.Text;
                //objOI.SelectedColor = txtSelectedColor.Text;
                //objOI.HeaderBackColor = txtHeaderBackColor.Text;
                //objOI.PagerBackColor = txtPagerBackColor.Text;

				try {
					objOI.ThumbColumns = int.Parse(txtThumbColumns.Text);
				}
				catch {
				}

                try
                {
                    objOI.ThumbRows = int.Parse(txtThumbRows.Text);
                }
                catch
                {
                }

				objOI.HideTextLink = chkHideTextLink.Checked;

				//Scroller Options
				objOI.ScrollBehavior = rblScrollBehavior.SelectedValue;
				objOI.ScrollDirection = rblScrollDirection.SelectedValue;
				objOI.ScrollAmount = txtScrollAmount.Text;
				objOI.ScrollDelay = txtScrollDelay.Text;
				objOI.ScrollWidth = txtScrollWidth.Text;
				objOI.ScrollHeight = txtScrollHeight.Text;

				//Audit Control
				objOI.UpdatedByUserId = UserInfo.UserID;
				objOI.UpdatedByUser = UserInfo.DisplayName;
				objOI.UpdatedDate = DateTime.Now;

				//.Save Options Info Object
				objOI.Update(ModId);
			}
		}

        private void AddUpdateSettingsItem()
        {
            try
            {
                //If Page.IsValid Then
                UpdateOptions(Convert.ToInt32(Request.QueryString["SettingsModuleId"]));
                // Redirect back to the portal home page
                Response.Redirect(Globals.NavigateURL(), true);
            }
            //End If
            catch (Exception exc)
            {
                //Module failed to load
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

		#endregion

		#region " Base Method Implementations "

		/// -----------------------------------------------------------------------------
		/// <summary>
		/// LoadSettings loads the settings from the Databas and displays them
		/// </summary>
		/// <remarks>
		/// </remarks>
		/// <history>
		///		[bhattji]	10/20/2004	created
		/// </history>
		/// -----------------------------------------------------------------------------
		public override void LoadSettings()
		{
			LoadOptions(ModuleId);
		}

		public void LoadOptions(int ModId)
		{
			try {
				if (!Page.IsPostBack) OptionsToPage(ModId); 
			}
			catch (Exception exc) {
				//Module failed to load
				Exceptions.ProcessModuleLoadException(this, exc);
			}
		}

		/// -----------------------------------------------------------------------------
		/// <summary>
		/// UpdateSettings saves the modified settings to the Database
		/// </summary>
		/// <remarks>
		/// </remarks>
		/// <history>
		///		[bhattji]	10/20/2004	created
		///		[bhattji]	10/25/2004	upated to use TabModuleId rather than TabId/ModuleId
		/// </history>
		/// -----------------------------------------------------------------------------
		public override void UpdateSettings()
		{
			UpdateOptions(ModuleId);
		}

		public void UpdateOptions(int ModId)
		{
			try {
				if (Page.IsValid) PageToOptions(ModId); 
			}
			catch (Exception exc) {
				//Module failed to load
				Exceptions.ProcessModuleLoadException(this, exc);
			}
		}

		#endregion

		#region " Event Methods "
		private void Page_Load(object sender, EventArgs e)
		{
			try {

				if (!IsPostBack)
				{
                    ulButtons.Visible = !string.IsNullOrEmpty(Request.QueryString["SettingsModuleId"]);
                    divPanelScript.Visible = ulButtons.Visible;
                    //Request.QueryString("SettingsModuleId") <> String.Empty
                    if (ulButtons.Visible)
					{
						LoadOptions(Convert.ToInt32(Request.QueryString["SettingsModuleId"]));

						SetIconBar();
					}
				}
			}
			catch (Exception exc) {
				//SettingsModule failed to load
				Exceptions.ProcessModuleLoadException(this, exc);
			}
		}

		private void cmdSetViewOrder_Click(object sender, EventArgs e)
		{
			XYZ70sController objXYZ70Controller = new XYZ70sController();
			objXYZ70Controller.SetViewOrderXYZ70s(ModuleId);
		}

		private void cmdClaimOrphan_Click(object sender, EventArgs e)
		{
			XYZ70sController objXYZ70Controller = new XYZ70sController();
			objXYZ70Controller.ClaimOrphanXYZ70s(ModuleId);
		}

		private void cmdSetViewOrderCat_Click(object sender, EventArgs e)
		{
			CategoriesController objCategoriesController = new CategoriesController();
			objCategoriesController.SetViewOrderCategories(ModuleId);
		}

		private void cmdClaimOrphanCat_Click(object sender, EventArgs e)
		{
			CategoriesController objCategoriesController = new CategoriesController();
			objCategoriesController.ClaimOrphanCategories(ModuleId);
		}

		/// -----------------------------------------------------------------------------
		/// <summary>
		/// cmdUpdate_Click runs when the update button is clicked
		/// </summary>
		/// <remarks>
		/// </remarks>
		/// <history>
		/// 	[bhattji]	9/23/2004	Updated to reflect design changes for Help, 508 support
		///                       and localisation
		/// </history>
		/// -----------------------------------------------------------------------------
		/// 

		private void imbUpdate_Click(object sender, ImageClickEventArgs e)
		{
			EventArgs cmdE = new EventArgs();
			cmdUpdate_Click(sender, cmdE);
		}
		private void cmdUpdate_Click(object sender, EventArgs e)
		{
            AddUpdateSettingsItem();
		}

		/// -----------------------------------------------------------------------------
		/// <summary>
		/// cmdCancel_Click runs when the cancel button is clicked
		/// </summary>
		/// <remarks>
		/// </remarks>
		/// <history>
		/// 	[bhattji]	9/23/2004	Updated to reflect design changes for Help, 508 support
		///                       and localisation
		/// </history>
		/// -----------------------------------------------------------------------------

		private void imbCancel_Click(object sender, ImageClickEventArgs e)
		{
			EventArgs cmdE = new EventArgs();
			cmdCancel_Click(sender, cmdE);
		}
		private void cmdCancel_Click(object sender, EventArgs e)
		{
			try {
				Response.Redirect(Globals.NavigateURL(), true);
			}
			catch (Exception exc) {
				//Module failed to load
				Exceptions.ProcessModuleLoadException(this, exc);
			}
		}

		#endregion

        #region " Web Form Designer generated code "
        override protected void OnInit(EventArgs e)
        {
            //
            // CODEGEN: This call is required by the ASP.NET Web Form Designer.
            //
            InitializeComponent();
            base.OnInit(e);
        }

        /// <summary>
        ///		Required method for Designer support - do not modify
        ///		the contents of this method with the code editor.
        /// </summary>
        /// 
        private void InitializeComponent()
        {
            //cmdSetViewOrder.Click += new EventHandler(cmdSetViewOrder_Click);
            //cmdClaimOrphan.Click += new EventHandler(cmdClaimOrphan_Click);
            //cmdSetViewOrderCat.Click += new EventHandler(cmdSetViewOrderCat_Click);
            //cmdClaimOrphanCat.Click += new EventHandler(cmdClaimOrphanCat_Click);
            //imbUpdate.Click += new ImageClickEventHandler(imbUpdate_Click);
            //cmdUpdate.Click += new EventHandler(cmdUpdate_Click);
            //imbCancel.Click += new ImageClickEventHandler(imbCancel_Click);
            //cmdCancel.Click += new EventHandler(cmdCancel_Click);
            ////this.Load += new EventHandler(this.Page_Load);
            cmdSetViewOrder.Click += cmdSetViewOrder_Click;
            cmdClaimOrphan.Click += cmdClaimOrphan_Click;
            cmdSetViewOrderCat.Click += cmdSetViewOrderCat_Click;
            cmdClaimOrphanCat.Click += cmdClaimOrphanCat_Click;
            //imbUpdate.Click += imbUpdate_Click;
            cmdUpdate.Click += cmdUpdate_Click;
            //imbCancel.Click += imbCancel_Click;
            cmdCancel.Click += cmdCancel_Click;
            //this.Load += Page_Load;
        }

        #endregion


	} //Settings

} //bhattji.Modules.XYZ70s

File: Jaydeep/DNN Training/JRC Modules/bhattji_Dispatchs/Settings.ascx.vb
' Copyright (c) 2002-2005
' by Jaydeep Bhatt, Vision Consultants. ( http://www.bhattji.com )
'
' Permission is hereby granted, to the person obtaining a copy of this software legaly and associated 
' documentation files (the "Software"), to deal in the Software with restriction, including with limitation 
' the rights to use, copy, modify, merge, PublishDate, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'

Imports System
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports DotNetNuke
Imports bhattji.Modules.Dispatchs.Business

Namespace bhattji.Modules.Dispatchs

    Public MustInherit Class Settings
        Inherits Entities.Modules.ModuleSettingsBase

#Region " Methods "

        Private Sub OptionsToPage(ByVal ModId As Integer)
            If ModId > 0 Then OptionsToPage(New OptionsInfo(ModId))
        End Sub

        Private Sub OptionsToPage(ByVal objOI As OptionsInfo)
            With objOI
                'Main Options
                Try
                    rblGetItems.SelectedIndex = .GetItems
                Catch
                    rblGetItems.SelectedIndex = 2
                End Try
                chkOnlyHostCanEdit.Checked = .OnlyHostCanEdit
                Try
                    rblViewControl.Items.FindByValue(.ViewControl).Selected = True
                Catch
                    rblViewControl.SelectedIndex = 0
                End Try
                txtPagerSize.Text = .PagerSize.ToString
                txtNoOfPages.Text = .NoOfPages.ToString
                chkDeleteFromGrid.Checked = .DeleteFromGrid

                Try
                    ddlUpdateRedirection.Items.FindByValue(.UpdateRedirection).Selected = True
                Catch
                End Try

                'Option1 Options
                chkAddDescription.Checked = .AddDescription
                Try
                    rblTabCss.Items.FindByValue(.TabCss).Selected = True
                Catch
                    rblTabCss.SelectedIndex = 0
                End Try
                Try
                    ddlTransition.Items.FindByValue(.Transition).Selected = True
                Catch
                    ddlTransition.Items.FindByValue("Pixelate").Selected = True
                End Try
                chkRattleImage.Checked = .RattleImage
                Try
                    ddlAddDate.Items.FindByValue(.AddDate).Selected = True
                Catch
                End Try
                Try
                    ddlImagePosition.Items.FindByValue(.ImagePosition).Selected = True
                Catch
                    ddlImagePosition.SelectedIndex = 1
                End Try

                chkDynamicThumb.Checked = .DynamicThumb
                chkShowRatings.Checked = .ShowRatings
                txtThumbWidth.Text = .ThumbWidth
                txtThumbHeight.Text = .ThumbHeight
                txtBackColor.Text = .BackColor
                txtAltColor.Text = .AltColor
                txtHeaderBackColor.Text = .HeaderBackColor
                txtPagerBackColor.Text = .PagerBackColor
                txtThumbColumns.Text = .ThumbColumns.ToString
                chkHideTextLink.Checked = .HideTextLink

                'Scroller Options
                Try
                    rblScrollBehavior.Items.FindByValue(.ScrollBehavior).Selected = True
                Catch
                    rblScrollBehavior.SelectedIndex = 0
                End Try
                Try
                    rblScrollDirection.Items.FindByValue(.ScrollDirection).Selected = True
                Catch
                    rblScrollDirection.SelectedIndex = 0
                End Try
                txtScrollAmount.Text = .ScrollAmount
                txtScrollDelay.Text = .ScrollDelay
                txtScrollWidth.Text = .ScrollWidth
                txtScrollHeight.Text = .ScrollHeight

                'Audit Control
                Try
                    'Dim objUC As New UserController
                    'Dim objUI As Users.UserInfo = Users.UserController.GetUser(PortalId, .UpdatedByUserId, True)
                    'ctlAudit.CreatedByUser = objUI.FirstName & " " & objUI.LastName
                    ctlAudit.CreatedByUser = .UpdatedByUser
                    ctlAudit.CreatedDate = .UpdatedDate.ToString
                    Dim objHostUser As Users.UserInfo = Users.UserController.GetUser(PortalId, UserId, True)
                    rblGetItems.Enabled = objHostUser.IsSuperUser
                    chkOnlyHostCanEdit.Enabled = objHostUser.IsSuperUser
                    If .OnlyHostCanEdit Then
                        'txtPayPalBusinessID.Enabled = objHostUser.IsSuperUser
                    End If
                Catch
                    ctlAudit.Visible = False
                End Try

            End With
        End Sub

        Private Sub PageToOptions(ByVal ModId As Integer)
            Dim objOI As New OptionsInfo
            With objOI
                'Main Options
                .GetItems = rblGetItems.SelectedIndex
                .OnlyHostCanEdit = chkOnlyHostCanEdit.Checked
                .ViewControl = rblViewControl.SelectedValue
                Try
                    .PagerSize = Integer.Parse(txtPagerSize.Text)
                Catch
                End Try
                Try
                    .NoOfPages = Integer.Parse(txtNoOfPages.Text)
                Catch
                End Try
                .DeleteFromGrid = chkDeleteFromGrid.Checked
                .UpdateRedirection = ddlUpdateRedirection.SelectedValue

                'Option1 Options
                .AddDescription = chkAddDescription.Checked
                .TabCss = rblTabCss.SelectedValue
                .Transition = ddlTransition.SelectedValue
                .RattleImage = chkRattleImage.Checked
                .AddDate = ddlAddDate.SelectedValue
                .ImagePosition = ddlImagePosition.SelectedValue
                .DynamicThumb = chkDynamicThumb.Checked
                .ShowRatings = chkShowRatings.Checked
                .ThumbWidth = txtThumbWidth.Text
                .ThumbHeight = txtThumbHeight.Text
                .BackColor = txtBackColor.Text
                .AltColor = txtAltColor.Text
                .HeaderBackColor = txtHeaderBackColor.Text
                .PagerBackColor = txtPagerBackColor.Text

                Try
                    .ThumbColumns = Integer.Parse(txtThumbColumns.Text)
                Catch
                End Try
                .HideTextLink = chkHideTextLink.Checked

                'Scroller Options
                .ScrollBehavior = rblScrollBehavior.SelectedValue
                .ScrollDirection = rblScrollDirection.SelectedValue
                .ScrollAmount = txtScrollAmount.Text
                .ScrollDelay = txtScrollDelay.Text
                .ScrollWidth = txtScrollWidth.Text
                .ScrollHeight = txtScrollHeight.Text

                'Audit Control
                .UpdatedByUserId = UserInfo.UserID
                .UpdatedByUser = UserInfo.DisplayName
                .UpdatedDate = Now

                '.Save Options Info Object
                .Update(ModId)
            End With
        End Sub

#End Region

#Region " Base Method Implementations "

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' LoadSettings loads the settings from the Databas and displays them
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        '''		[bhattji]	10/20/2004	created
        ''' </history>
        ''' -----------------------------------------------------------------------------
        ''' 
        Public Overrides Sub LoadSettings()
            LoadOptions(ModuleId)
        End Sub

        Public Sub LoadOptions(ByVal ModId As Integer)
            Try
                If Not Page.IsPostBack Then OptionsToPage(ModId)
            Catch exc As Exception    'Module failed to load
                ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' UpdateSettings saves the modified settings to the Database
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        '''		[bhattji]	10/20/2004	created
        '''		[bhattji]	10/25/2004	upated to use TabModuleId rather than TabId/ModuleId
        ''' </history>
        ''' -----------------------------------------------------------------------------
        Public Overrides Sub UpdateSettings()
            UpdateOptions(ModuleId)
        End Sub

        Public Sub UpdateOptions(ByVal ModId As Integer)
            Try
                If Page.IsValid Then PageToOptions(ModId)
            Catch exc As Exception    'Module failed to load
                ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

#End Region

#Region " Event Methods "
        Private Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Try
                'Dim ModId As Integer = ModuleId
                'If Request.QueryString("SettingsModuleId") Is Nothing Then
                '    ModId = Convert.ToInt16(Request.QueryString("SettingsModuleId"))
                'End If

                'Dim objOI As New OptionsInfo(ModId)
                'If objOI.OnlyHostCanEdit Then
                '    Dim objUC As New UserController
                '    Dim objHostUser As UserInfo = objUC.GetUser(PortalId, UserId)
                '    If Not objHostUser.IsSuperUser Then
                '        ' security violation attempt to access item not related to this Module
                '        'DotNetNuke.UI.Skins.Skin.AddModuleMessage(Me, "<a href=""" & NavigateURL() & """ title=""Click to go back"" onmouseover""window.status=this.title; return true"">Requires Host rights</a>", DotNetNuke.UI.Skins.Controls.ModuleMessage.ModuleMessageType.RedError)
                '        Response.Redirect(NavigateURL(), True)
                '    End If
                'End If

                If Not IsPostBack Then
                    cmdUpdate.Visible = Not Request.QueryString("SettingsModuleId") Is Nothing 'Request.QueryString("SettingsModuleId") <> String.Empty
                    imbUpdate.Visible = cmdUpdate.Visible
                    cmdCancel.Visible = cmdUpdate.Visible
                    imbCancel.Visible = cmdCancel.Visible

                    If cmdUpdate.Visible Then
                        LoadOptions(Convert.ToInt16(Request.QueryString("SettingsModuleId")))
                    End If
                End If
            Catch exc As Exception    'SettingsModule failed to load
                ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

        Private Sub cmdSort_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdSort.Click
            Dim objDispatchController As New DispatchsController
            objDispatchController.SortDispatchs(ModuleId)
        End Sub

        Private Sub cmdFix_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdFix.Click
            Dim objDispatchController As New DispatchsController
            objDispatchController.FixDispatchs(ModuleId, UserId)
        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' cmdUpdate_Click runs when the update button is clicked
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[bhattji]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------
        ''' 

        Private Sub imbUpdate_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles imbUpdate.Click
            Dim cmdE As New EventArgs
            cmdUpdate_Click(sender, cmdE)
        End Sub

        Private Sub cmdUpdate_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdUpdate.Click
            Try
                'If Page.IsValid Then
                UpdateOptions(Convert.ToInt16(Request.QueryString("SettingsModuleId")))
                ' Redirect back to the portal home page
                Response.Redirect(NavigateURL(), True)
                'End If
            Catch exc As Exception    'Module failed to load
                ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' cmdCancel_Click runs when the cancel button is clicked
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[bhattji]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------

        Private Sub imbCancel_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles imbCancel.Click
            Dim cmdE As New EventArgs
            cmdCancel_Click(sender, cmdE)
        End Sub

        Private Sub cmdCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdCancel.Click
            Try
                Response.Redirect(NavigateURL(), True)
            Catch exc As Exception    'Module failed to load
                ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

#End Region

    End Class 'Settings

End Namespace 'bhattji.Modules.Dispatchs
